### ISSUE TITLE
issue with calculating the loss while using GPT-2

### ISSUE THREAD

**Issue by dahani9091** [KimMeen/Time-LLM#41]
I'm using an Nvidia v100 GPU. I have changed the --mixed_precision to FP16 instead of BF16 since this GPU card can't work on  BF16. I have also adjusted this line `enc_out, n_vars = self.patch_embedding(x_enc.to(torch.bfloat16))` in TimeLLM.py
to be:
`enc_out, n_vars = self.patch_embedding(x_enc.to(torch.float16))`

I'm running this on GPT-2

Everything went well, but during the training phase I'm getting **nan** losses. Please see the attached screenshot.

![image](https://github.com/KimMeen/Time-LLM/assets/69170574/a54f6304-2b2b-4960-a30e-04088c6cac82)

Any thoughts why this is happening ?

**Comment by kwuking** [comment]
hi 

> BF16
Hi, may I ask which dataset you are training on? Is it possible to train using A100 and BF16 precision?

**Comment by kwuking** [comment]
> I'm using an Nvidia v100 GPU. I have changed the --mixed_precision to FP16 instead of BF16 since this GPU card can't work on BF16. I have also adjusted this line `enc_out, n_vars = self.patch_embedding(x_enc.to(torch.bfloat16))` in TimeLLM.py to be: `enc_out, n_vars = self.patch_embedding(x_enc.to(torch.float16))`
> 
> I'm running this on GPT-2
> 
> Everything went well, but during the training phase I'm getting **nan** losses. Please see the attached screenshot.
> 
> ![image](https://private-user-images.githubusercontent.com/69170574/314267753-a54f6304-2b2b-4960-a30e-04088c6cac82.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA5MDQ5OTQsIm5iZiI6MTcxMDkwNDY5NCwicGF0aCI6Ii82OTE3MDU3NC8zMTQyNjc3NTMtYTU0ZjYzMDQtMmIyYi00OTYwLWEzMGUtMDQwODhjNmNhYzgyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzIwVDAzMTgxNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTkxYTEyNWIxZmM5ZTNhMTNkMWQ4YTQ5MDI5M2RmMzg2YzU3NDI4NWI5OTY5NDRiM2U5ZWJjMjljYzM0ZmRjMDEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Xd3OsNVS_OtpgM2oaNG-WrgRkb0xtMA0dwhu73POoto)
> 
> Any thoughts why this is happening ?

Hi, may I ask which dataset you are training on? Is it possible to train using A100 and BF16 precision?

**Comment by dahani9091** [comment]
Update about this issue:
solved.
Solution: decreased start learning rate to 0.0001
